### DISCUSSION TITLE
Proposal for enabling group buy on ONDC

### DISCUSSION THREAD

**Discussion by BLR-0118** [ONDC-Official/protocol-network-extension#43] [General]
Proposal for enabling group buy on ONDC is [here](https://docs.google.com/document/d/1Gg1Lu4FA_EqQkmz-V8_9kpC9ZDV3J0WCUmmCzCdw-uw/edit#heading=h.mdesd9mpd29m). Request participants to comment before we can take it forward.
@dashbitla-ib @aakasjayaraj @abaweja @batramanuj @mahoriR @mherle @VidyaGF @NajeebMohammed https://github.com/orgs/ONDC-Official/teams/participants

**Comment by dashbitla-ib** [comment]
@BLR-0118 

This may not be a great use case and too much of a change to the contract and not backward compatible. Group Buy as a feature - I didn't see it hugely successful anywhere in the leading commerce platforms. Infact it delays the decision to buy by individuals as many a times people do impulse purchases as they found something interesting and just click to buy.

This is something the Buyer Apps can offer if they want to even with the current construct by doing multiple Orders.

I suggest few important things to focus on for the next version:

**Optimize the Search Responses**

-       make it very light weight - the data bills are going to be huge for both seller and buyer platforms with too much of metadata for each row of data. Do similar to CSV like data and merge all flags into bit strings to shorten the data
-       make it almost cache free way of building Buyer platforms. A very light weight API to show the Search Results like data. An API to pull the data for a selective item ids. the buyer apps who cache, provide a mechanism to refresh their cached data as it changes at the Seller. A simple changed Item Ids to be posted to the Buyer Plaforms and an API to pull the Items Data for the selected items based on the Item Ids

**Faster Checkout Flows / Single Click Buy Feature**

-     anything that we can do to implement this!?

**Offers Module**

-       item level offers
-       order level offers
-       BXGX / BOGO
-       Volume Based Offers - that helps in implementing Group Buy Exp
